### DISCUSSION TITLE
Pre-processing visual inspection of artifacts

### DISCUSSION THREAD

**Discussion by MarcoSleepCBT-I** [raphaelvallat/yasa#58] [Q&A] [answered]
Hi @raphaelvallat,
thank you very much for your great work!
I am using your library, and everything works well. On the other hand, I would visually inspect EEG traces to reject artifacts 30-sec epochs in EDFbrowser and then run your library. I would import the hypnogram with epochs as you require: 0 = Wake, 1 = N1 sleep, 2 = N2 sleep, 3 = N3 sleep and 4 = REM sleep, but also with an annotation named 6 = Artifact.
How can I implement this without issues in your automatic artifact rejection alghorithm (artifact_rejection()) and other features (spindles_detection(), spindles_slow_fast: (), sw_detection: (), bandpower(), spectrogram())?

Thank you again,

Marco

**Comment by raphaelvallat** [comment, marked as answer]
Hi @MarcoSleepCBT-I,

In YASA, Artefacts are denoted -1 in the hypnogram. Most of the time, -1 epochs will just be ignored by YASA's function (though they will be shown in plot_spectrogram). Just to be clear, are you scoring the artefacts manually in EDFBrowser, or using the automatic artefact rejection tool in YASA?

Thanks,
Raphael

**Comment by MarcoSleepCBT-I** [comment]
Hi @raphaelvallat,

I would like to score the 30-sec artifacts first manually in EDFBrowser, and afterward use the automatic artifact rejection tool in YASA. Is it possible? If I denote manually the 30-sec artifacts with "-1" in the EDFbrowser's annotations, and then run your YASA's functions, could there be problems?

Thank you,
Marco

**Reply by raphaelvallat** [reply]
This is definitely possible, the hardest part is probably to load the EDFBrowser annotations into a YASA hypnogram format (i.e. 1 value for each 30-sec epoch: `hypno = [0, 0, 1, 1, -1, 2, .... -1, -1]`). Once you've figured this out, just run all the YASA functions as usual and the -1 epochs will be ignored.

**Reply by MarcoSleepCBT-I** [reply]
Thank you very much @raphaelvallat,

I converted the hypnogram (annotations) from EDFbrowser format into YASA format simply with Notepad++, and I labeled the Artifact epochs as "-1".
Everything in YASA functions worked well!

Thank you again,

Marco
